File: python/questions/rightSideView.py
from typing import Optional, List

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        def io(n: Optional[TreeNode], d: int) -> None:
            if n is None:
                return
            if d > len(ans):
                ans.append(n.val)
            io(n.right, d+1)
            io(n.left, d+1)
        io(root, 1)
        return ans

File: cpp/questions/rightSideView.cpp
#include <bits/stdc++.h>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        std::vector<int> ans;
        std::function<void(TreeNode*, int)> io = [&](TreeNode* n, int d) -> void {
            if (n == nullptr) {
                return;
            }
            if (d > ans.size()) {
                ans.push_back(n->val);
            }
            io(n->right, d+1);
            io(n->left, d+1);
        };
        io(root, 1);
        return ans;
    }
};
